File: app/src/main/java/com/sapuglha/espressotips/HelloActivity.java
package com.sapuglha.espressotips;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import butterknife.BindView;
import butterknife.ButterKnife;

public class HelloActivity extends AppCompatActivity {

    @BindView(R.id.hello_activity_text_name)
    TextView name;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_hello);

        ButterKnife.bind(this);

        Intent intent = getIntent();
        Bundle bundle = intent.getExtras();
        if (null != bundle) {
            String param = (String) bundle.get("name");
            name.setText(param);
        }
    }
}

File: app/src/main/java/com/sapuglha/espressotips/HelloActivity.kt
package com.sapuglha.espressotips

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_hello.*

class HelloActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_hello)

        val bundle = intent?.extras
        bundle?.let {
            val param = bundle.get("name") as String
            hello_activity_text_name.text = param
        }
    }
}